packages/flame/lib/src/components/component.dart

```diff
@@ -736,6 +759,7 @@ class Component {
     assert(_parent!.findGame()!.hasLayout);
     _setLoadingBit();
     onGameResize(_parent!.findGame()!.canvasSize);
+    _updateInheritance();
```

Comment:
Is this the right place to update inheritance? Does this cover change parent use cases? 

I will soon add tests to cover this scenarios.